tests/extra_source_and_tests/user_2d_T_shaped_pipe_VIPO_rigid_shell/T_shaped_pipe_VIPO_rigid_shell.cpp

```diff
@@ -0,0 +1,444 @@
+/**
+ * @file 	T_shaped_pipe.cpp
+ * @brief 	This is the benchmark test of multi-inlet and multi-outlet.
+ * @details We consider a flow with one inlet and two outlets in a T-shaped pipe in 2D.
+ * @author 	Xiangyu Hu, Shuoguo Zhang
+ */
+
+#include "sphinxsys.h"
+#include "bidirectional_buffer.h"
+#include "density_correciton.h"
+#include "density_correciton.hpp"
+#include "kernel_summation.h"
+#include "kernel_summation.hpp"
+#include "pressure_boundary.h"
+
+using namespace SPH;
+//----------------------------------------------------------------------
+//	Basic geometry parameters and numerical setup.
+//----------------------------------------------------------------------
+Real DL = 0.2;                                               /**< Reference length. */
+Real DH = 0.1;                                               /**< Reference and the height of main channel. */
+Real DL1 = 0.75 * DL;                                        /**< The length of the main channel. */
+Real resolution_ref = 0.002;                                 /**< Initial reference particle spacing. */
+Real BW = resolution_ref * 1.0;                                
+Real buffer_width = resolution_ref * 4.0;                                /**< Reference size of the emitter. */
+Real DL_sponge = resolution_ref * 20;                        /**< Reference size of the emitter buffer to impose inflow condition. */
+StdVec<Vecd> observer_location = {Vecd(0.5 * DL, 0.5 * DH)}; /**< Displacement observation point. */
+Real level_set_refinement_ratio = resolution_ref / (0.1 * BW);
+//----------------------------------------------------------------------
+//	Global parameters on the fluid properties
+//----------------------------------------------------------------------
+Real Outlet_pressure = 0.1;
+Real rho0_f = 1060;                                                 /**< Reference density of fluid. */
+Real Re = 100.0;                                                      /**< Reynolds number. */
+Real U_f = 0.5;                                                      /**< Characteristic velocity. */
+Real mu_f = 0.00355; /**< Dynamics viscosity. */
+Real c_f =10.0 * U_f * SMAX(Real(1), DH / (Real(2.0) * (DL - DL1)));
```

Comment:
From the output files U_max in the flow field is about 0.85, make sure c > 10 * U_max
```
Real U_max = 1.2 * 1.5 * U_f * SMAX(Real(1), DH / (Real(2.0) * (DL - DL1))); /**< Maximum fluid velocity. */
Real c_f = 10.0 * U_max;
```

Reply:
Properties of the fluid have been wrongly applied here and have been corrected according to the case test_2d_modified_T_flow.